deepchecks/core/check.py

```diff
@@ -458,6 +459,79 @@ def name(cls):
         return split_camel_case(cls.__name__)
 
 
+class SingleDatasetBaseCheck(BaseCheck):
+    """Parent class for checks that only use one dataset."""
+
+    context_type: ClassVar[Optional[Type[Any]]] = None  # TODO: Base context type
+
+    def __init__(self):
+        super().__init__()
+        # Replace the run_logic function with wrapped run function
+        setattr(self, 'run_logic', wrap_run(getattr(self, 'run_logic'), self))
+
+    def run(self, dataset, model=None) -> CheckResult:
+        """Run check."""
+        assert self.context_type is not None
+        return self.run_logic(self.context_type(  # pylint: disable=not-callable
+            dataset,
+            model=model
+        ))
+
+    @abc.abstractmethod
+    def run_logic(self, context, dataset_type: str = 'train') -> CheckResult:
+        """Run check."""
+        raise NotImplementedError()
+
+
+class TrainTestBaseCheck(BaseCheck):
+    """Parent class for checks that compare two datasets.
+
+    The class checks train dataset and test dataset for model training and test.
+    """
+
+    context_type: ClassVar[Optional[Type[Any]]] = None  # TODO: Base context type
+
+    def __init__(self):
+        super().__init__()
+        # Replace the run_logic function with wrapped run function
+        setattr(self, 'run_logic', wrap_run(getattr(self, 'run_logic'), self))
+
+    def run(self, train_dataset, test_dataset, model=None) -> CheckResult:
+        """Run check."""
+        assert self.context_type is not None
+        return self.run_logic(self.context_type(  # pylint: disable=not-callable
+            train_dataset,
+            test_dataset,
+            model=model
+        ))
+
+    @abc.abstractmethod
+    def run_logic(self, context) -> CheckResult:
+        """Run check."""
+        raise NotImplementedError()
+
+
+class ModelOnlyBaseCheck(BaseCheck):
+    """Parent class for checks that only use a model and no datasets."""
+
+    context_type: ClassVar[Optional[Type[Any]]] = None  # TODO: Base context type
+
+    def __init__(self):
+        super().__init__()
+        # Replace the run_logic function with wrapped run function
+        setattr(self, 'run_logic', wrap_run(getattr(self, 'run_logic'), self))
+
+    def run(self, model) -> CheckResult:
```

Comment:
why we switched the run and run_logic?

Reply:
1. take a look at suite impl, it calls run_logic  ` check_result = check.run_logic(context) `
2. I did not change it was like this